test/helper/e2e2/yml/yml.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2025 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yml
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/scale/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/api/v1"
+	akov2next "github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/nextapi/v1"
+)
+
+var (
+	// ErrNoCR indicates the parsed YAML is not valid CR
+	ErrNoCR = errors.New("YAML definition is not a CR")
+)
+
+type autoCloser struct {
+	io.ReadCloser
+	closed bool
+}
+
+func (ac *autoCloser) Read(b []byte) (int, error) {
+	if ac.closed {
+		return 0, io.EOF
+	}
+	n, err := ac.ReadCloser.Read(b)
+	if err == io.EOF {
+		if err := ac.ReadCloser.Close(); err != nil {
+			log.Printf("autoCloser failed to close %v: %v", ac.ReadCloser, err)
+		}
+	}
+	return n, err
+}
+
+func MustOpen(fsys fs.FS, path string) io.Reader {
+	f, err := fsys.Open(path)
+	if err != nil {
+		panic(fmt.Errorf("Fatal: could not open virtual file system path %q: %w", path, err))
+	}
+	return &autoCloser{ReadCloser: f}
+}
+
+func MustParseCRs(ymls io.Reader) []client.Object {
+	objs, err := ParseCRs(ymls)
+	if err != nil {
+		panic(fmt.Errorf("Fatal: could not parse CRs: %w", err))
+	}
+	return objs
+}
+
+func ParseCRs(ymls io.Reader) ([]client.Object, error) {
```

Comment:
s/ParseCRs/ParseObject

"CR" is not a thing in kube source code lingua. You are parsing a client object, hence let's call the function appropriately.